src/app/common/LoadingBlock.vue

```diff
@@ -19,6 +14,7 @@
 </template>
 
 <script lang="ts" setup>
-import { KUI_ICON_SIZE_50 } from '@kong/design-tokens'
-import { KEmptyState, KIcon } from '@kong/kongponents'
+import { KEmptyState } from '@kong/kongponents'
+
+import ProgressIcon from './ProgressIcon.vue'
```

Comment:
question: is this the wrapper you have (for animation) around the `@kong/icons` ProgressIcon? 

I'm wondering if there's a way we can add the animation into the icon itself with inline styles so that it doesn't require a stylesheet import 🤔

Reply:
> question: is this the wrapper you have (for animation) around the @kong/icons ProgressIcon?

Yes, for reference, this is the entire component.

```vue
<template>
  <ProgressIcon
    class="progess-icon"
    display="inline-block"
    :color="KUI_COLOR_TEXT_NEUTRAL_WEAK"
  />
</template>

<script lang="ts" setup>
import { KUI_COLOR_TEXT_NEUTRAL_WEAK } from '@kong/design-tokens'
import { ProgressIcon } from '@kong/icons'
</script>

<style lang="scss">
.progress-icon path {
  transform-origin: center;
  animation: spin 1s linear infinite;
}

@keyframes spin {
  100% {
    transform:rotate(360deg);
  }
}
</style>
```

> I'm wondering if there's a way we can add the animation into the icon itself with inline styles so that it doesn't require a stylesheet import 🤔

Not sure how we can do this. We can’t put `@keyframes` in inline styles, so that won’t work with what I have here. We could utilize https://developer.mozilla.org/en-US/docs/Web/SVG/Element/animate but that would need to live in `@kong/icons` as we would need to change the SVG content.

Reply:
Just a thought, but could the animation be done in the SVG itself? I'm guessing we always want the progress icon to be spinning? (apart from `@media prefers-reduced-motion` maybe?)

Reply:
Resolving this for the moment because this isn’t something we can solve on our side.

Reply:
With SVG `<animate>`, we could do something like this in Kong/icons:

`svg/solid/progress.svg`:
```svg
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M12 22C10.6333 22 9.34167 21.7375 8.125 21.2125C6.90833 20.6875 5.84583 19.9708 4.9375 19.0625C4.02917 18.1542 3.3125 17.0917 2.7875 15.875C2.2625 14.6583 2 13.3667 2 12C2 10.6167 2.2625 9.32083 2.7875 8.1125C3.3125 6.90417 4.02917 5.84583 4.9375 4.9375C5.84583 4.02917 6.90833 3.3125 8.125 2.7875C9.34167 2.2625 10.6333 2 12 2C12.2833 2 12.5208 2.09583 12.7125 2.2875C12.9042 2.47917 13 2.71667 13 3C13 3.28333 12.9042 3.52083 12.7125 3.7125C12.5208 3.90417 12.2833 4 12 4C9.78333 4 7.89583 4.77917 6.3375 6.3375C4.77917 7.89583 4 9.78333 4 12C4 14.2167 4.77917 16.1042 6.3375 17.6625C7.89583 19.2208 9.78333 20 12 20C14.2167 20 16.1042 19.2208 17.6625 17.6625C19.2208 16.1042 20 14.2167 20 12C20 11.7167 20.0958 11.4792 20.2875 11.2875C20.4792 11.0958 20.7167 11 21 11C21.2833 11 21.5208 11.0958 21.7125 11.2875C21.9042 11.4792 22 11.7167 22 12C22 13.3667 21.7375 14.6583 21.2125 15.875C20.6875 17.0917 19.9708 18.1542 19.0625 19.0625C18.1542 19.9708 17.0958 20.6875 15.8875 21.2125C14.6792 21.7375 13.3833 22 12 22Z" fill="black" transform-origin="center" />
<animateTransform attributeName="transform" attributeType="XML" type="rotate" from="0" to="360" dur="1.5s" repeatCount="indefinite" />
</svg>
```

The downside of using SVG `<animate>` is, as far as I can tell anyway, that we wouldn’t be able to use a prefers-reduced-motion query. However, that also doesn’t work with only using inline styles.

Reply:
I’ve created a PR over in Kong/icons to discuss this further: https://github.com/Kong/icons/pull/82

Reply:
```html
<span style="width: 40px;height: 40px;display: block;background-color: red;-webkit-mask-image: url('data:image/svg+xml,%3Csvg fill=%27none%27 height=%27100%25%27 viewBox=%270 0 24 24%27 width=%27100%25%27 xmlns=%27http://www.w3.org/2000/svg%27%3E%3Cstyle%3E @keyframes spin %7B 100%25 %7B transform:rotate(360deg); %7D %7D path %7B transform-origin: center; animation: spin 1s linear infinite; %7D %3C/style%3E%3Cpath d=%27M12 22C10.6333 22 9.34167 21.7375 8.125 21.2125C6.90833 20.6875 5.84583 19.9708 4.9375 19.0625C4.02917 18.1542 3.3125 17.0917 2.7875 15.875C2.2625 14.6583 2 13.3667 2 12C2 10.6167 2.2625 9.32083 2.7875 8.1125C3.3125 6.90417 4.02917 5.84583 4.9375 4.9375C5.84583 4.02917 6.90833 3.3125 8.125 2.7875C9.34167 2.2625 10.6333 2 12 2C12.2833 2 12.5208 2.09583 12.7125 2.2875C12.9042 2.47917 13 2.71667 13 3C13 3.28333 12.9042 3.52083 12.7125 3.7125C12.5208 3.90417 12.2833 4 12 4C9.78333 4 7.89583 4.77917 6.3375 6.3375C4.77917 7.89583 4 9.78333 4 12C4 14.2167 4.77917 16.1042 6.3375 17.6625C7.89583 19.2208 9.78333 20 12 20C14.2167 20 16.1042 19.2208 17.6625 17.6625C19.2208 16.1042 20 14.2167 20 12C20 11.7167 20.0958 11.4792 20.2875 11.2875C20.4792 11.0958 20.7167 11 21 11C21.2833 11 21.5208 11.0958 21.7125 11.2875C21.9042 11.4792 22 11.7167 22 12C22 13.3667 21.7375 14.6583 21.2125 15.875C20.6875 17.0917 19.9708 18.1542 19.0625 19.0625C18.1542 19.9708 17.0958 20.6875 15.8875 21.2125C14.6792 21.7375 13.3833 22 12 22Z%27 fill=%27currentColor%27%3E%3C/path%3E%3C/svg%3E')"></span>
```
:see_no_evil: 😅